internal/benchmark/metrics/consensus/latency.go

```diff
@@ -20,19 +21,22 @@ const (
 
 type LatencyMetric struct {
 	metric.Base[time.Duration]
-	url       string
-	interval  time.Duration
-	durations []time.Duration
+	url               *url.URL
+	interval, timeout time.Duration
+	durations         []time.Duration
 }
 
-func NewLatencyMetric(url, name string, interval time.Duration, healthCondition []metric.HealthCondition[time.Duration]) *LatencyMetric {
+func NewLatencyMetric(addr, name string, interval time.Duration, healthCondition []metric.HealthCondition[time.Duration]) *LatencyMetric {
+	url, _ := url.Parse(addr)
```

Comment:
dropping errors that you expect not to fail is bad practise, it leads to hard-to-find issues when code changes

Reply:
made some refactor for better error handling